File: packages/appwsgi/appwsgi%201050.tar.bz2/appwsgi/www/vi/vi.js
// Copyright(c) gert.cuykens@gmail.com
vi=
{
 'read':function()
 {
  http.fml=function(xml)
  {
   if (xml.getElementsByTagName('error')[0]) alert(xml.getElementsByTagName('error')[0].childNodes[0].nodeValue)
   document.getElementById('text').value=xml.getElementsByTagName('root')[0].childNodes[0].data
  }
  http.xml ='<?xml version="1.0" encoding="UTF-8"?>\n'
  http.xml+='<root>\n'
  http.xml+=' <cmd>read</cmd>\n'
  http.xml+=' <gid>admin</gid>\n'
  var p='vi.txt'; try{p=document.URL.match(/\?(.*)/)[1]} catch(e){}
  http.xml+=' <path>'+p+'</path>\n'
  http.xml+='</root>'
  http.url='vi.py'
  http.send()
 },

 'write':function()
 {
  http.fml=function(xml)
  {
   if (xml.getElementsByTagName('error')[0]) alert(xml.getElementsByTagName('error')[0].childNodes[0].nodeValue)
   document.getElementById('text').value=xml.getElementsByTagName('root')[0].childNodes[0].data
  }
  http.xml ='<?xml version="1.0" encoding="UTF-8"?>\n'
  http.xml+='<root>\n'
  http.xml+=' <cmd>write</cmd>\n'
  http.xml+=' <gid>admin</gid>\n'
  var p='vi.txt'; try{p=document.URL.match(/\?(.*)/)[1]} catch(e){}
  http.xml+=' <path>'+p+'</path>\n'
  http.xml+=' <txt>'+document.getElementById('text').value+'</txt>\n'
  http.xml+='</root>'
  http.url='vi.py'
  http.send()
 }
}

File: packages/appwsgi/appwsgi%201050.tar.bz2/appwsgi/www/vi/vi.py
# Copyright(c) gert.cuykens@gmail.com
from frame import Frame
from session import Session
from db import Db
from os import path

def application(environ, response):
    cookie = environ.get('HTTP_COOKIE','')
    document = environ['wsgi.input'].read(int(environ.get('CONTENT_LENGTH','0')))
    db = Db()
    v = Frame.read(document)
    s = Session(db,cookie,v['gid'])
    
    response('200 OK', [('Content-type', 'text/xml')])
    xml  = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
    xml += "<root>"

    if not s.GID == 'admin':
        xml += " <error>Access denied!</error>\n"
        xml += "</root>"
        return [xml]

    if v['cmd'] == 'write':
        f = open(path.join(path.dirname(__file__),'vi.txt'), 'w')
        f.write(v['txt'])
        f.close()

    xml += "<![CDATA["

    with open(path.join(path.dirname(__file__),'vi.txt'), 'r') as f:
         for line in f:
             xml += line

    xml += "]]>"
    xml += "</root>"

    return [xml]

